.github/workflows/src/summarize-checks.js

```diff
@@ -0,0 +1,854 @@
+    troubleshootingGuide: defaultTsg,
+  },
+  {
+    precedence: 1,
+    name: "Swagger PrettierCheck",
+    suppressionLabels: [],
+    troubleshootingGuide: defaultTsg,
+  },
+  {
+    precedence: 2,
+    name: "Swagger SemanticValidation",
+    suppressionLabels: [],
+    troubleshootingGuide: defaultTsg,
+  },
+  {
+    precedence: 3,
+    name: "Swagger ModelValidation",
+    suppressionLabels: [],
+    troubleshootingGuide: defaultTsg,
+  },
+  {
+    precedence: 4,
+    name: "Swagger BreakingChange",
+    suppressionLabels: [verRevApproval, brChRevApproval],
+    troubleshootingGuide: brchTsg,
+  },
+  {
+    precedence: 4,
+    name: "Breaking Change(Cross-Version)",
+    suppressionLabels: [verRevApproval, brChRevApproval],
+    troubleshootingGuide: brchTsg,
+  },
+  {
+    precedence: 5,
+    name: "Swagger LintDiff",
+    suppressionLabels: [],
+    troubleshootingGuide: defaultTsg,
+  },
+  {
+    precedence: 5,
+    name: "Swagger Lint(RPaaS)",
+    suppressionLabels: [],
+    troubleshootingGuide: defaultTsg,
+  },
+  {
+    precedence: 6,
+    name: "SDK azure-sdk-for-net",
+    suppressionLabels: [],
+    troubleshootingGuide: checkAndDiagramTsg(3),
+  },
+  {
+    precedence: 6,
+    name: "SDK azure-sdk-for-net-track2",
+    suppressionLabels: [],
+    troubleshootingGuide: checkAndDiagramTsg(3),
+  },
+  {
+    precedence: 6,
+    name: "SDK azure-sdk-for-go",
+    suppressionLabels: [],
+    troubleshootingGuide: checkAndDiagramTsg(3),
+  },
+  {
+    precedence: 6,
+    name: "SDK azure-sdk-for-java",
+    suppressionLabels: [],
+    troubleshootingGuide: checkAndDiagramTsg(3),
+  },
+  {
+    precedence: 6,
+    name: "SDK azure-sdk-for-js",
+    suppressionLabels: [],
+    troubleshootingGuide: checkAndDiagramTsg(3),
+  },
+  {
+    precedence: 6,
+    name: "SDK azure-sdk-for-python",
+    suppressionLabels: [],
+    troubleshootingGuide: checkAndDiagramTsg(3),
+  },
+  {
+    precedence: 6,
+    name: "SDK azure-sdk-for-python-track2",
+    suppressionLabels: [],
+    troubleshootingGuide: checkAndDiagramTsg(3),
+  },
+  {
+    precedence: 10,
+    name: AUTOMATED_CHECK_NAME,
+    suppressionLabels: [],
+    troubleshootingGuide: reqMetCheckTsg,
+  },
+];
+
+// during renderAutomatedMergingRequirementsMetCheck we resolve the result of
+// automated merge requirements met by from the result of and(requiredChecks).
+// if any are pending, automated merging requirements is pending. This is ripe for complete removal
+// in favor of just honoring the `required` checks results directly.
+/** @type {string[]} */
+const EXCLUDED_CHECK_NAMES = [];
+
+// #endregion
+// #region core
+/**
+ * @param {import('github-script').AsyncFunctionArguments} AsyncFunctionArguments
+ * @returns {Promise<void>}
+ */
+export default async function summarizeChecks({ github, context, core }) {
+  logGitHubRateLimitInfo({ github, context, core });
+  let { owner, repo, issue_number, head_sha } = await extractInputs(github, context, core);
```

Comment:
I have a code clarity question for you @mikeharder . Here, we obviously return the github values of `issue_number` and `head_sha` because that's the naming convention in github.

Would you prefer that I maintain this naming convention as I flow these values throughout the script? OR would you prefer I map them to the intended js-appropriate name. Eg `issue_number` -> `prNumber`. `head_sha` -> `sha`. `event_name` -> `eventName`.

Reply:
This is a tricky problem.  You can look at existing code like "arm-auto-signoff.js".  I think it's fairly consistent, but I didn't create explicit naming rules for this.

I think my pattern was basically:

1. If we're interacting with the GH APIs, use `snake_case`
2. Else, use `camelCase`

One thing I did try to do, is use the same words but just different case.  So `issue_number` and `issueNumber` should mean the same thing, but I wouldn't assume `prNumber` and `issue_number` are the same.

https://github.com/Azure/azure-rest-api-specs/blob/517eaf1cca58813605768f4ddc9a59ca75173493/.github/workflows/src/arm-auto-signoff.js#L49-L54

I don't think it's worth nitpicking this either way.